tests/test_logging.py

```diff
@@ -131,18 +131,26 @@ def test_log_message_gets_formatted(self):
 
         assert result['message'] == "hello bar"
 
-    def test_log_message_is_unchanged_if_fields_are_not_found(self):
+    def test_log_message_shows_missing_key1_if_fields_are_not_found(self):
         self.logger.info("hello {bar}")
         result = json.loads(self.buffer.getvalue())
 
-        assert result['message'] == "hello {bar}"
+        assert result['message'] == "hello {bar: missing key}"
 
-    def test_failed_log_message_formatting_logs_an_error(self):
+    def test_missing_key_when_formatting_logs_a_warning(self):
         self.logger.info("hello {barry}")
         raw_result = self.dmbuffer.getvalue()
         result = json.loads(raw_result)
 
-        assert result['message'].startswith("failed to format log message")
+        assert result['message'].startswith("Missing keys when formatting log message: ['barry']")
```

Comment:
Could we get a test for when there are two missing keys please so it's explicit what we expect when more than 1 missing key